### ISSUE TITLE
Extensions report available updates from pre-release channel when running released extensions

### ISSUE THREAD

**Issue by kwmlodozeniec** [microsoft/vscode#214474]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.90
- OS Version: Ubuntu 22.04

Steps to Reproduce:

1. Install latest released Python Debugger extension.
2. Restart VSCode and check for extension updates.
3. Sometimes, not always, updates are available but they are obviously from the pre-release channel.
![image](https://github.com/microsoft/vscode/assets/1453791/05dd899a-0135-431b-a4f1-eea87ab2bd0d)

**Comment by sandy081** [comment]
I do not see this extension version at all. Could be this extension version was published as release version by mistake? Are you still seeing this?

**Comment by kwmlodozeniec** [comment]
Yes, I am still seeing this. It seems related to an extension being in preview but having a release and pre-release channels.

**Comment by sandy081** [comment]
> It seems related to an extension being in preview but having a release and pre-release channels.

Not sure if I understand this. 

Have you opted to install pre-release version of this extension when you installed it?

**Comment by kwmlodozeniec** [comment]
No, I did not opt in to use a pre-release version. I mentioned preview as it might be relevant. One of the other extensions that exhibit this issue is in preview, so I thought it might be relevant.

**Comment by kwmlodozeniec** [comment]
Just to make it clear, the suggested update version is not released in the release channel but in the pre-release.

**Comment by sandy081** [comment]
Strange. In the screenshot you have shared, there is also no button to Switch To Pre-Release or Switch To Release. What do you see when you perform **Install Another Version...* action?

**Comment by kwmlodozeniec** [comment]
I have this problem with three extensions: Python Debugger, Black and isort.
![black](https://github.com/microsoft/vscode/assets/1453791/1bf18a86-970d-472b-8a91-5c8061513725)
![debug](https://github.com/microsoft/vscode/assets/1453791/a347c0a5-d3e9-464a-b075-dc7eba65c932)
![isort](https://github.com/microsoft/vscode/assets/1453791/afbc2216-10bb-42d2-8226-48895783a0d5)

Using the debugger extension as example. After I update it I get this, with a clear option to switch to release version
![debug_after_update](https://github.com/microsoft/vscode/assets/1453791/e655cd5b-edc2-42ed-a742-7ef7bea65ee6)

After I switch to the release version I get this, with an option to switch to pre-release version
![debug_after_switch_to_release](https://github.com/microsoft/vscode/assets/1453791/fa0c0ee1-d9df-42a4-a8b0-6ad8e7591ac9)

After this that extension no longer reports the pre-release update as available when I am on the released version. But, I have done this before and it seems to eventually come back.

**Comment by sandy081** [comment]
> But, I have done this before and it seems to eventually come back.

Ok, that's strange. Please have an eye on this and report with the steps if you see it again. I am not sure if there is anything I can do without knowing how you got into this state.

**Comment by kwmlodozeniec** [comment]
Indeed. It seems it was some weird cache issue and the fact that my settings were syncing between different platforms, mainly MacOS and Linux. This is what I did that seemed to have resolved it for me:

- Purged all cache on both systems 
- Switched problematic extensions to pre-release channel 
- Switched problematic extensions to release - channel
- Ensure the option to "Show pre-release versions" is disabled

I have not seen the issue for over a week now so I think I got it fixed.
